File: api/app/schemas/player.py
"""Player-related response schemas."""

from pydantic import BaseModel

from app.schemas.grade import GradeBrief, PlayerGradeSummary


class PlayerSearchItem(BaseModel):
    """A player search result."""

    player_id: int
    player_name: str
    position: str | None
    position_group: str | None
    nationality: str | None


class PlayerSearchResponse(BaseModel):
    """Player search results."""

    players: list[PlayerSearchItem]


class PlayerTransfer(BaseModel):
    """A transfer in the player's history."""

    transfer_id: int
    from_club_id: int
    from_club_name: str
    to_club_id: int
    to_club_name: str
    fee_eur: int | None
    fee_is_loan: bool
    transfer_date: str | None
    transfer_window: str
    season: str
    grade: GradeBrief | None = None


class PlayerValuationPoint(BaseModel):
    """A single market valuation data point."""

    date: str
    value_eur: int


class PlayerDetailResponse(BaseModel):
    """Full player profile."""

    player_id: int
    name: str
    date_of_birth: str | None
    position: str | None
    position_group: str | None
    nationality: str | None
    transfermarkt_url: str | None
    transfers: list[PlayerTransfer]
    valuations: list[PlayerValuationPoint]
    grade_summary: PlayerGradeSummary | None = None

File: web/src/types/player.ts
/** A player search result. */
export interface PlayerSearchResult {
  player_id: number;
  player_name: string;
  position: string | null;
  position_group: string | null;
  nationality: string | null;
}

/** A transfer in the player's career. */
export interface PlayerTransfer {
  transfer_id: number;
  from_club_id: number;
  from_club_name: string;
  to_club_id: number;
  to_club_name: string;
  fee_eur: number | null;
  fee_is_loan: boolean;
  transfer_date: string | null;
  transfer_window: string;
  season: string;
}

/** A market valuation data point. */
export interface PlayerValuation {
  date: string;
  value_eur: number;
}

/** Full player profile. */
export interface PlayerDetail {
  player_id: number;
  name: string;
  date_of_birth: string | null;
  position: string | null;
  position_group: string | null;
  nationality: string | null;
  transfermarkt_url: string | null;
  transfers: PlayerTransfer[];
  valuations: PlayerValuation[];
}
